src/Compilers/VisualBasic/Test/Semantic/SourceGeneration/GeneratorDriverTests_Attributes_FullyQualifiedName.vb

```diff
@@ -0,0 +1,1030 @@
+                                                                                                                              End Sub)
+                                                                                          End Sub))
+
+            Dim driver As GeneratorDriver = VisualBasicGeneratorDriver.Create(ImmutableArray.Create(Of ISourceGenerator)(generator), parseOptions:=parseOptions, driverOptions:=New GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps:=True))
+            driver = driver.RunGenerators(compilation)
+            Dim runResult = driver.GetRunResult().Results(0)
+
+            Assert.Collection(runResult.TrackedSteps("result_ForAttributeWithMetadataName"),
+                Sub(_step) Assert.True(IsClassStatementWithName(_step.Outputs.Single().Value, "C1")))
+        End Sub
+
+        <Fact>
+        Public Sub FindCorrectAttributeOnTopLevelClass_WhenSearchingForClassDeclaration2()
+            Dim source = "
+<N1.X>
+class C1
+end class
+<N2.X>
+class C2
+end class
+
+namespace N1
+    class XAttribute
+        inherits System.Attribute
+    end class
+end namespace
+
+namespace N2
+    class XAttribute
+        inherits System.Attribute
+    end class
+end namespace
+"
+            Dim parseOptions = TestOptions.RegularLatest
+            Dim compilation As Compilation = CreateCompilation(source, options:=TestOptions.DebugDll, parseOptions:=parseOptions)
+
+            Assert.Single(compilation.SyntaxTrees)
+
+            Dim generator = New IncrementalGeneratorWrapper(New PipelineCallbackGenerator(Sub(ctx)
+                                                                                              Dim input = ctx.ForAttributeWithMetadataName(Of ClassStatementSyntax)("N2.XAttribute")
+                                                                                              ctx.RegisterSourceOutput(input, Sub(spc, node)
+                                                                                                                              End Sub)
+                                                                                          End Sub))
+
+            Dim driver As GeneratorDriver = VisualBasicGeneratorDriver.Create(ImmutableArray.Create(Of ISourceGenerator)(generator), parseOptions:=parseOptions, driverOptions:=New GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps:=True))
+            driver = driver.RunGenerators(compilation)
+            Dim runResult = driver.GetRunResult().Results(0)
+
+            Assert.Collection(runResult.TrackedSteps("result_ForAttributeWithMetadataName"),
+Sub(_step) Assert.True(IsClassStatementWithName(_step.Outputs.Single().Value, "C2")))
+        End Sub
+
+        <Fact>
+        Public Sub DoNotAttributeOnTopLevelClass_WhenSearchingForSimpleName1()
```

Comment:
Consider using a `<Theory>` here

```suggestion
        <Theory>
        <InlineData("XAttribribute")>
        <InlineData("X")>
        Public Sub DoNotAttributeOnTopLevelClass_WhenSearchingForSimpleName1(name as String)
```

Then us `name` as the search item in line 162. This should allow you to combine several of the unit tests together